### PR TITLE
TokenRing: allow passing an external one into the app

### PR DESCRIPTION
**Pull request by lukaszdudek-silvair**
That will simplify usage in our gateway application.

### CODE REVIEW

File: bluetooth_mesh/application.py [hunk 1]
```diff
@@ -670,12 +670,12 @@ async def attach(self, token: Optional[int] = None):
         Returns current node configuration, see documentation for Attach()
         method in mesh-api.txt_.
         """
-        token = token if token is not None else self.token_ring.token
+        token = token or self.token_ring.token
```
Comment (michallowasrzechonek-silvair) [inline bluetooth_mesh/application.py:673]: token can be `0`

File: bluetooth_mesh/application.py [hunk 2]
```diff
@@ -419,7 +419,7 @@ class Application(
 
     ELEMENTS = {}  # type: Dict[int, Type[Element]]
 
-    def __init__(self, loop: asyncio.AbstractEventLoop):
+    def __init__(self, loop: asyncio.AbstractEventLoop, token_ring: Optional[TokenRing] = None):
```
Comment (michallowasrzechonek-silvair) [inline bluetooth_mesh/application.py:422]: How about making this a mixin?

```
class TokenRingMixin:
    TOKEN_RING = TokenRing

    @property
    def token_ring(self):
        reutn self.TOKEN_RING()
```

then you can just change the class when deriving from `Application`

Comment (michallowasrzechonek-silvair) [reply bluetooth_mesh/application.py:422]: My inspiration was http://ccbv.co.uk/, from my XP it works wonders.

Comment (lukaszdudek-silvair) [reply bluetooth_mesh/application.py]: See the current version.

**Pull request merged**